### PR TITLE
Cleanup and password tests

### PR DESCRIPTION
**Pull request by DFurnes**
#### Changes
I'm trying to figure out why some users aren't validating when we check Phoenix logins against Northstar... haven't found any big issues yet, but did figure out that we're not removing the `drupal_password` field when a user changes their password on Phoenix. (525627f)

I added some more unit tests with known good/bad Drupal password hashes as well (acc164c).

Finally, an unrelated change: I moved shared controller methods into traits, rather than having them all on the base `Controller` class. This makes it a bit easier to skim, and will make it easier to yank them out into a shared package at some point. (cd1e643)

#### How should this be reviewed?
I'd recommend going commit-by-commit. :1234: 

#### How should this be tested?
:traffic_light: 

---
For review: @angaither @weerd

### CODE REVIEW

File: tests/PasswordTest.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        ]);
+
+        // Assert user has been updated in the database with a newly hashed password.
+        $user = $user->fresh();
+        $this->assertArrayNotHasKey('drupal_password', $user['attributes']);
+        $this->assertArrayHasKey('password', $user['attributes']);
+
+        // Finally, let's try logging in with the newly hashed password
+        $this->withScopes(['user'])->json('POST', 'v1/auth/verify', [
+            'email' => 'dries.buytaert@example.com',
```
Comment (angaither) [inline tests/PasswordTest.php]: totes profesh

File: tests/PasswordTest.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        $this->assertResponseStatus(200);
+    }
+
+    /**
+     * Test that updating a user's password will re-hash it.
+     */
+    public function testUpdatingUser()
+    {
+        $user = User::create([
+            'email' => 'acquia.consultant@example.com',
```
Comment (angaither) [inline tests/PasswordTest.php]: yikes

**Comment by angaither** [comment]
SURE

File: tests/PasswordTest.php (hunk 1 above)
Comment (DFurnes) [inline tests/PasswordTest.php]: :sweat_drops:

File: tests/PasswordTest.php (hunk 2 above)
Comment (DFurnes) [inline tests/PasswordTest.php]: :wink:

**Pull request merged**